backend/src/test/java/org/eclipse/tractusx/puris/backend/demandandcapacitynotification/logic/adapter/DemandAndCapacityNotificationSammMapperTest.java

```diff
@@ -0,0 +1,540 @@
+        Assertions.assertEquals(jsonNode.get("status").asText(), StatusEnumeration.OPEN.getValue());
+
+        Assertions.assertTrue(jsonNode.get("materialGlobalAssetId").isArray());
+        for (var field: jsonNode.get("materialGlobalAssetId")) {
+            Assertions.assertEquals(CX_MAT_NUMBER, field.asText());
+        }
+
+        Assertions.assertTrue(jsonNode.get("materialNumberSupplier").isArray());
+        for (var field: jsonNode.get("materialNumberSupplier")) {
+            Assertions.assertEquals(SUPPLIER_MAT_NUMBER, field.asText());
+        }
+
+        Assertions.assertTrue(jsonNode.get("materialNumberCustomer").isArray());
+        for (var field: jsonNode.get("materialNumberCustomer")) {
+            Assertions.assertEquals(CUSTOMER_MAT_NUMBER, field.asText());
+        }
+    }
+
+    @ParameterizedTest
+    @ValueSource(strings = {regularSammFromSupplier, sammWithoutCXIdFromSupplier, sammWithoutCXIdAndSupplierMnrFromSupplier})
+    void testSammDeSerializationAsCustomer(String receivedSammString) throws Exception {
+        // parse a Samm, we received from supplier as a customer
+
+        DemandAndCapacityNotificationSamm samm = objectMapper.readValue(receivedSammString, DemandAndCapacityNotificationSamm.class);
+        Partner mySelf = customerPartner;
+        Partner externalPartner = supplierPartner;
+
+        Material semiconductorMaterial = new Material(true, false,
+            CUSTOMER_MAT_NUMBER, null, "Semiconductor");
+
+        MaterialPartnerRelation materialPartnerRelation = new MaterialPartnerRelation(semiconductorMaterial, externalPartner,
+            SUPPLIER_MAT_NUMBER, true, false);
+
+        when(materialService.findByOwnMaterialNumber(CUSTOMER_MAT_NUMBER)).thenReturn(semiconductorMaterial);
+        when(mprService.findByPartnerAndPartnerCXNumber(externalPartner, CX_MAT_NUMBER)).thenReturn(materialPartnerRelation);
+        when(mprService.find(semiconductorMaterial, externalPartner)).thenReturn(materialPartnerRelation);
+        when(partnerService.getOwnPartnerEntity()).thenReturn(mySelf);
+
+        ReportedDemandAndCapacityNotification reportedDemandAndCapacityNotification =
+            demandAndCapacityNotificationSammMapper.sammToReportedDemandAndCapacityNotification(samm, externalPartner);
+        LOG.info(reportedDemandAndCapacityNotification::toString);
+
+        Assertions.assertEquals(1, reportedDemandAndCapacityNotification.getMaterials().size());
+        Assertions.assertEquals(semiconductorMaterial, reportedDemandAndCapacityNotification.getMaterials().getFirst());
+        Assertions.assertEquals(StatusEnumeration.OPEN, reportedDemandAndCapacityNotification.getStatus());
+        Assertions.assertEquals(LeadingRootCauseEnumeration.LOGISTICS_DISRUPTION, reportedDemandAndCapacityNotification.getLeadingRootCause());
+        Assertions.assertEquals(EffectEnumeration.CAPACITY_REDUCTION, reportedDemandAndCapacityNotification.getEffect());
+    }
```

Comment:
Please add assertions for material numbers. Currently we have checks only for serialization but not for deserialization.

Reply:
The test asserts, that the ReportedDemandAndCapacityNotification entity, we receive from the Samm Mapper contains the expected material entity (and no other material). see lines 205, 206. 

The ReportedDemandAndCapacityNotification in itself does not contain any material numbers, so there is nothing to assert here. The material numbers only exist as members of the material entities / mpr entities. 

Perhaps some more explanation, what this test method is actually meant to do, might be helpful: 

The "testSammDeSerializationAsCustomer" is focused on showing, whether the material identification logic in the DemandAndCapacityNotificationSammMapper (lines 97 - 121) works as expected. 

The SAMM's usually contain redundat information on the affected materials, since they (usually) not only contain a the CX-id but also the customer-/supplier materialnumbers. 

This test is repeated three times with different samm inputs. While the first input "regularSammFromSupplier" contains fully redundant information, the other two tests cases have partially missing information ("sammWithoutCXIdFromSupplier", "sammWithoutCXIdAndSupplierMnrFromSupplier")

Reply:
Thanks for elaborating. I actually missed that.